test/parallel/test-http-writable-true-after-close.js

```diff
@@ -7,29 +7,35 @@ const { get, createServer } = require('http');
 // res.writable should not be set to false after it has finished sending
 // Ref: https://github.com/nodejs/node/issues/15029
 
+let internal;
 let external;
 
-// Http server
-const internal = createServer((req, res) => {
-  res.writeHead(200);
-  setImmediate(common.mustCall(() => {
-    external.abort();
-    res.end('Hello World\n');
-  }));
-}).listen(0);
-
 // Proxy server
 const server = createServer(common.mustCall((req, res) => {
   get(`http://127.0.0.1:${internal.address().port}`, common.mustCall((inner) => {
-    res.on('close', common.mustCall(() => {
+    // on CentOS, 'finish' is emitted
+    res.on('finish', () => {
       assert.strictEqual(res.writable, true);
-    }));
+    });
+    // everywhere else, 'close' is emitted
+    res.on('close', () => {
+      assert.strictEqual(res.writable, true);
+    });
```

Comment:
This will test less than before, since the `mustCall` is not in place anymore. It might be worth to check that either `finish` or `close` will be called at least once? And I am somewhat surprised that our events are platform dependent. Would you be so kind and add a comment why that is the case?

Reply:
I have no idea on why it's the case, that's the major problem of this PR. 

I'll see if I can get another solution as well.

I'll add another check that verifies that at least one of them is called.